### PR TITLE
swarm/storage: Support for uploading 100gb files

### LINKED ISSUE #1356: Support for uploading 100gb files
This is an umbrella issue for fixing all the issues related to large file uploads.

### PR DESCRIPTION
**Pull request by jmozah**
Fixes #1356 

WIP: Please don't merge yet

### CODE REVIEW

File: swarm/storage/hasherstore.go [hunk 1]
```diff
@@ -26,15 +26,23 @@ import (
 	"golang.org/x/crypto/sha3"
 )
 
+const (
+	NoOfStorageWorkers = 150
```
Comment (acud) [inline swarm/storage/hasherstore.go]: This is not used anywhere else outside the package, so does not need to be exported. also, please rename to `maxStorageWorkers` or similar

File: swarm/storage/hasherstore.go [hunk 2]
```diff
@@ -26,15 +26,23 @@ import (
...
+)
+
 type hasherStore struct {
 	store     ChunkStore
 	toEncrypt bool
+	frstChunk bool
```
Comment (acud) [inline swarm/storage/hasherstore.go]: rename to `firstChunk`

File: swarm/storage/hasherstore.go [hunk 3]
```diff
@@ -81,6 +92,11 @@ func (h *hasherStore) Put(ctx context.Context, chunkData ChunkData) (Reference,
 	chunk := h.createChunk(c)
 	h.storeChunk(ctx, chunk)
 
+	if !h.frstChunk {
```
Comment (acud) [inline swarm/storage/hasherstore.go]: why not use `sync.Once` instead?

File: swarm/storage/localstore/localstore.go [hunk 4]
```diff
@@ -44,7 +44,7 @@ var (
 
 var (
 	// Default value for Capacity DB option.
-	defaultCapacity uint64 = 5000000
+	defaultCapacity uint64 = 30000000
```
Comment (acud) [inline swarm/storage/localstore/localstore.go:47]: revert back to default `5000000`. this implicates all swarm users

File: swarm/storage/localstore/localstore.go [hunk 5]
```diff
@@ -134,7 +134,7 @@ func New(path string, baseKey []byte, o *Options) (db *DB, err error) {
 	if o == nil {
 		// default options
 		o = &Options{
-			Capacity: 5000000,
+			Capacity: 30000000,
```
Comment (acud) [inline swarm/storage/localstore/localstore.go]: use the `const` above instead of hardcoding the number

**Review by acud — CHANGES_REQUESTED** [review]
nice one @jmozah. please see a few comments in the body of the PR

File: swarm/storage/hasherstore.go (hunk 1 above)
Comment (jmozah) [reply swarm/storage/hasherstore.go]: Done

File: swarm/storage/hasherstore.go (hunk 2 above)
Comment (jmozah) [reply swarm/storage/hasherstore.go]: Removed this and added sync.once

File: swarm/storage/hasherstore.go (hunk 3 above)
Comment (jmozah) [reply swarm/storage/hasherstore.go]: Done

File: swarm/storage/localstore/localstore.go (hunk 4 above)
Comment (jmozah) [reply swarm/storage/localstore/localstore.go:47]: isn't our intention is to support 100gb for all users? Today we support only 20gb.

File: swarm/storage/localstore/localstore.go (hunk 5 above)
Comment (jmozah) [reply swarm/storage/localstore/localstore.go]: Done

**Comment by acud** [comment]
@jmozah please rebase - this is conflicting as we merged the rc branch to master

**Comment by jmozah** [comment]
Rebasing from master.. so closing this PR.

**Pull request closed without merging**